Settings controller [MAILPOET-1757]

costasovo opened the pull request:
I'm very sorry for such a looooong PR. Most of it are just replacements of static calls.
It should be reviewable when going commit by commit

After this gets merged we have to use this controller instead of static Setting::getValue/setValue.
This has to be rebased and checked/upadted for some new occurrences of Setting::xxx calls before merging.

pavel-mailpoet commented on lib/Settings/SettingsController.php:
```diff
@@ -0,0 +1,104 @@
+      }
+    }
+    $setting[$last_key] = $value;
+    Setting::setValue($main_key, $settings[$main_key]);
```
Should we move `Setting::setValue` and `Setting::getValue` to here to discourage people from using it?

costasovo replied:
Very good point!

pavel-mailpoet commented on lib/Config/Populator.php:
```diff
@@ -154,13 +157,13 @@ private function createDefaultSettings() {
     // enable signup confirmation by default
-    if(!Setting::getValue('signup_confirmation')) {
-      Setting::setValue('signup_confirmation', array(
+    if(!$this->settings->fetch('signup_confirmation')) {
```
I don't understand why do we need to use `fetch` in this class.

pavel-mailpoet commented on lib/Settings/SettingsController.php:
```diff
@@ -0,0 +1,104 @@
+    $keys = explode('.', $key);
+    $main_key = $keys[0];
+    $last_key = array_pop($keys);
+    $settings = self::$settings;
```
I don't think we need this variable here.

pavel-mailpoet commented on lib/Newsletter/ViewInBrowser.php:
```diff
@@ -2,12 +2,19 @@
+  /** @var bool */
+  private $is_tracking_enabled;
+
+  function __construct($is_tracking_enabled) {
```
How is this working? I don't see passing the argument anywhere.

costasovo replied:
It is passed here: https://github.com/mailpoet/mailpoet/pull/1766/commits/e7dfdf1cdbfd354ed1084a53f4d1f13f6b6b711e#diff-a565a70a1dc7bccd9aec5961b86160f9R35

costasovo commented:
I've moved handling defaults into new controller and also moved static setValue, getValue as a private methods to the controller so that they can be used any more...
also rebased on current master

costasovo commented:
Rebased on current master
